File: backend/core/src/main/java/com/feed_grabber/core/email/MailContentBuilder.java
package com.feed_grabber.core.email;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.thymeleaf.TemplateEngine;
import org.thymeleaf.context.Context;

@Service
public class MailContentBuilder {
    private final TemplateEngine templateEngine;

    @Autowired
    public MailContentBuilder(TemplateEngine templateEngine) {
        this.templateEngine = templateEngine;
    }

    public String buildInviteHtmlForm(String link) {
        Context context = new Context();
        context.setVariable("link", link);
        return templateEngine.process("inviteLinkTemplate/index", context);
    }
}

File: backend/event_processor/src/main/kotlin/com/feed_grabber/event_processor/email/MailContentBuilder.kt
package com.feed_grabber.event_processor.email;

import com.sendgrid.helpers.mail.objects.Content
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.thymeleaf.TemplateEngine
import org.thymeleaf.context.Context

@Service
class MailContentBuilder(
        @Autowired val templateEngine: TemplateEngine
) {

    fun buildAccountActivationMail(link: String): Content {
        val context = Context()
        context.setVariable("link", link)
        return Content("text/html", templateEngine.process("activate_account", context))
    }

}
